EventsExpress/ClientApp/src/containers/event-list.js

```diff
@@ -1,49 +1,41 @@
 import React, { Component } from 'react';
+import { getFormValues } from 'redux-form';
 import { connect } from 'react-redux';
 import { parse as queryStringParse } from 'query-string';
 import EventList from '../components/event/event-list';
 import Spinner from '../components/spinner';
 import { get_events } from '../actions/event/event-list-action';
-import history from '../history';
 import eventHelper from '../components/helpers/eventHelper';
+import { withRouter } from "react-router";
 
 class EventListWrapper extends Component {
     constructor(props) {
         super(props);
         this.objCurrentQueryParams = Object.create(null);
+        this.prevQueryStringSearch = "";
     }
 
     componentDidMount() {
-        this.setSearchParamsToEventFilter(this.props.location.search);
-        this.executeSearchEvents();
+        this.setSearchParamsToEventFilter(this.props.history.location.search);
+        const queryString = eventHelper.getQueryStringByEventFilter(this.objCurrentQueryParams);
+        this.props.get_events(queryString);
     }
 
-    componentDidUpdate(prevProps) {
-        const objFilterParams = eventHelper.trimUndefinedKeys(this.props.events.filter);
-        if (this.hasUpdateSearchParams(objFilterParams)) {
-            this.objCurrentQueryParams = objFilterParams;
-            this.executeSearchEvents();
+    componentDidUpdate() {
+        if (window.location.search != this.prevQueryStringSearch) {
```

Comment:
Can we use connect this component to router and use search property from there?

Reply:
only instead of window.location.search. PrevProps I cannot use in this case, they are equal to this.props